src/metabase/search/postgres/index.clj

```diff
@@ -155,23 +155,29 @@
     (when @reindexing?
       (t2/delete! pending-table :model_id id :model [:in search-models]))))
 
-(defn- process-negation [term]
-  (if (str/starts-with? term "-")
-    (str "!" (subs term 1))
-    term))
+(defn- quote* [s]
+  (str "'" (str/replace s "'" "''") "'"))
```

Comment:
The apostrophes get ignored in the lexems anyway, but perhaps this is tweakable.